### PR TITLE
Test that the swiftinterface for the stdlib can be compiled

### PR DESCRIPTION
**Pull request by jrose-apple**
...like we've done with the overlays for months. Meant to do this a long time ago, but at the time we hadn't gotten the stdlib working yet!

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

File: validation-test/ParseableInterface/verify_stdlib.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Note that this test should still "pass" when Swift.swiftinterface has not
+// been generated.
+
+// RUN: %empty-directory(%t)
+// RUN: test ! -e %platform-module-dir/Swift.swiftmodule/%target-cpu.swiftinterface || %target-swift-frontend -build-module-from-parseable-interface %platform-module-dir/Swift.swiftmodule/%target-cpu.swiftinterface -parse-stdlib -o %t/Swift.swiftmodule
```
Comment (harlanhaskins) [inline validation-test/ParseableInterface/verify_stdlib.swift]: @compnerd Is this good on Windows?

**Review by harlanhaskins — APPROVED** [review]
🎉

**Review by nathawes — APPROVED** [review]

File: validation-test/ParseableInterface/verify_stdlib.swift (hunk 1 above)
Comment (jrose-apple) [reply validation-test/ParseableInterface/verify_stdlib.swift]: Oops, right. I copied from `verify_simd` but that's guarded by `objc_interop` anyway.

**Comment by jrose-apple** [comment]
@swift-ci Please test Linux

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Pull request merged**